### PR TITLE
Documention on how we handle authentication

### PR DESCRIPTION
**Pull request by krak3n**
This PR adds documentation on how we do authentication using the standard `Authorization` header.

@peeklondon / @edwardoparearyee / @radeklos / @jamesjwarren another one for feedback :stuck_out_tongue:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -497,7 +497,73 @@ So an example maybe:
...
+in the form of `{id}:{session_id}` where `id` is a hash of the user `id` and `session_id` is
+a unique Session ID used to validate the user.
+
+### Example Flow
+
+* First `POST` credentials to the appropriate authentication service, here we will use
+  `auth.service` which would be an internal micro service handling user authentication.
+
+``` http
+GET / HTTP/1.1
```
Comment (edoparearyee) [inline README.md]: should this be `POST`?

Comment (krak3n) [inline README.md]: Fixed

**Comment by edoparearyee** [comment]
:+1:

**Comment by jamesjwarren** [comment]
:+1:

**Comment by krak3n** [comment]
@peeklondon / @radeklos does this sound sane to you guys as well?

**Comment by radeklos** [comment]
:+1:

**Pull request merged**